packages/@aws-cdk/aws-stepfunctions/lib/fields.ts

```diff
@@ -39,6 +39,11 @@ export class Data {
     return new JsonPathToken('$').toString();
   }
 
+  public static isJsonPath(value: any): boolean {
+    const decoded = Token.decode(value);
```

Comment:
This can use that unexported function from `json-path.ts`.  That already does what you want, no need to add `Token.decode()`.

And in fact, you could implement this in terms of `findReferencedJsonPaths(value).length > 0` (or whatever the function is again) at the call site, so no need to make this function at all.